samples/sample_multi_transcode/src/pipeline_transcode.cpp

```diff
@@ -149,78 +149,12 @@ CTranscodingPipeline::CTranscodingPipeline():
     shouldUseGreedyFormula(false),
     m_nRotationAngle(0)
 {
-    MSDK_ZERO_MEMORY(m_mfxDecParams);
-    MSDK_ZERO_MEMORY(m_mfxVppParams);
-    MSDK_ZERO_MEMORY(m_mfxEncParams);
-    MSDK_ZERO_MEMORY(m_mfxPluginParams);
-    MSDK_ZERO_MEMORY(m_RotateParam);
-    MSDK_ZERO_MEMORY(m_mfxPreEncParams);
-
-    MSDK_ZERO_MEMORY(m_threadsPar);
-    MSDK_ZERO_MEMORY(m_initPar);
-
-    MSDK_ZERO_MEMORY(m_mfxDecResponse);
-    MSDK_ZERO_MEMORY(m_mfxEncResponse);
-
-    MSDK_ZERO_MEMORY(m_Request);
-
-    MSDK_ZERO_MEMORY(m_VppDoNotUse);
-    MSDK_ZERO_MEMORY(m_MVCSeqDesc);
-    MSDK_ZERO_MEMORY(m_EncOpaqueAlloc);
-    MSDK_ZERO_MEMORY(m_VppOpaqueAlloc);
-    MSDK_ZERO_MEMORY(m_DecOpaqueAlloc);
-    MSDK_ZERO_MEMORY(m_PluginOpaqueAlloc);
-    MSDK_ZERO_MEMORY(m_PreEncOpaqueAlloc);
-    MSDK_ZERO_MEMORY(m_ExtLAControl);
-    MSDK_ZERO_MEMORY(m_AvcTemporalLayers);
-    MSDK_ZERO_MEMORY(m_CodingOptionSPSPPS);
-    MSDK_ZERO_MEMORY(m_CodingOption);
-    MSDK_ZERO_MEMORY(m_CodingOption2);
-    MSDK_ZERO_MEMORY(m_CodingOption3);
-    MSDK_ZERO_MEMORY(m_ExtHEVCParam);
-    MSDK_ZERO_MEMORY(m_ExtHEVCTiles);
-#if MFX_VERSION >= 1022
-    MSDK_ZERO_MEMORY(m_decPostProcessing);
-    m_decPostProcessing.Header.BufferId = MFX_EXTBUFF_DEC_VIDEO_PROCESSING;
-    m_decPostProcessing.Header.BufferSz = sizeof(mfxExtDecVideoProcessing);
-#endif //MFX_VERSION >= 1022
-
-    m_MVCSeqDesc.Header.BufferId = MFX_EXTBUFF_MVC_SEQ_DESC;
-    m_MVCSeqDesc.Header.BufferSz = sizeof(mfxExtMVCSeqDesc);
-
-    m_VppDoNotUse.Header.BufferId = MFX_EXTBUFF_VPP_DONOTUSE;
-    m_VppDoNotUse.Header.BufferSz = sizeof(mfxExtVPPDoNotUse);
-
-    m_ExtHEVCParam.Header.BufferId = MFX_EXTBUFF_HEVC_PARAM;
-    m_ExtHEVCParam.Header.BufferSz = sizeof(mfxExtHEVCParam);
-
-    m_ExtHEVCTiles.Header.BufferId = MFX_EXTBUFF_HEVC_TILES;
-    m_ExtHEVCTiles.Header.BufferSz = sizeof(mfxExtHEVCTiles);
-
-#if (MFX_VERSION >= 1026)
-    MSDK_ZERO_MEMORY(m_ExtVP9Param);
-    m_ExtVP9Param.Header.BufferId = MFX_EXTBUFF_VP9_PARAM;
-    m_ExtVP9Param.Header.BufferSz = sizeof(mfxExtVP9Param);
-#endif
-
-#if (MFX_VERSION >= 1024)
-    MSDK_ZERO_MEMORY(m_ExtBRC);
-    m_ExtBRC.Header.BufferId = MFX_EXTBUFF_BRC;
-    m_ExtBRC.Header.BufferSz = sizeof(m_ExtBRC);
-#endif
-
-#if (MFX_VERSION >= 1025)
-
-    MSDK_ZERO_MEMORY(m_ExtMFEParam);
-    MSDK_ZERO_MEMORY(m_ExtMFEControl);
-
-    m_ExtMFEControl.Header.BufferId = MFX_EXTBUFF_MULTI_FRAME_CONTROL;
-    m_ExtMFEControl.Header.BufferSz = sizeof(mfxExtMultiFrameControl);
+    m_RotateParam = {};
```

Comment:
I'd move every trivial initialization in the constructor body to initialization list above

Reply:
Dmitry R. agains